### ISSUE TITLE
CSS selectors compiled wrong

### ISSUE THREAD

**Issue by krisdigital** [sass/sassc-rails#60]
When updating to the newest Rails version 4.2.5.1, sassc (1.8.4) and sassc-rails (1.2.0) were also updated. Suddenly Bootstrap is compiled with strange errors. For example input-groups:

````
.input-group .form-control:not(:first-child, :first-child, :first-child):not(:last-child, :last-child, :last-child)
````

The :first-child and :last-child statements should obviously only appear once and let the statement break in Chrome and Firefox. Switching back to sass-rails solves the error.

Any idea what change caused this?

**Comment by bolandrm** [comment]
Try sassc-ruby 1.8.3.  This may be a bug in the latest version of libsass.

**Comment by krisdigital** [comment]
@bolandrm Thank's for the suggestion. Just tried it and does not make a difference!

**Comment by dsteppeler** [comment]
Hi!
I am interested in a fix for this as well. Currently it is preventing me from using this gem.
It seems to be related to this one: https://github.com/sass/libsass/issues/1901

Regards!

**Comment by samjewell** [comment]
I'd also love a fix for this.
We saw it around checkboxes and radio buttons
```
:not(:checked)
```
 was compiled to
```
:not(:checked, :checked)
```

**Comment by bolandrm** [comment]
I've just released sassc 1.8.5 which upgrades libsass to 3.3.4.  Please give that a try!

**Comment by ElManouche** [comment]
I've now sassc 1.9.0, but the same issue.
Any other suggestions?

**Comment by ElManouche** [comment]
I have more details after some tests.
The error only appears when the parent selector has a period as well:
`.test {
  // this works
  .test__elem {
    &:not(.active) {
      opacity: 0.5;
    }
  }

  // this creates a bug in the 'not' selector
  .test__item,
  .test__item2 {
    &:not(.active) {
      opacity: 0.5;
    }
  }
}`

is compiled to:
`.test .test__elem:not(.active) {
  opacity: 0.5;
}

.test .test__item:not(.active, .active),
.test .test__item2:not(.active, .active) {
  opacity: 0.5;
}`

**Comment by bolandrm** [comment]
@ElManouche could you please post this bug here:  https://github.com/sass/libsass/issues  and link your issue back to this thread so we can track it?

**Comment by xzyfer** [comment]
@bolandrm this is fixed in 3.3.4.

**Comment by bolandrm** [comment]
@xzyfer i thought so... sassc 1.8.5 / 1.9.0 are using libsass 3.3.4, so this _should_ be fixed.
